ConverterViewController.swift

```diff
@@ -0,0 +1,174 @@
+            if error != nil
+            {
+                print("error=\(error)")
+                return
+            }
+            
+            // Print out response string
+            let responseString = NSString(data: data!, encoding: NSUTF8StringEncoding)
+            print("responseString = \(responseString)")
+            
+            
+            // Convert server json response to NSDictionary
+            do {
+                if let convertedJsonIntoDict = try NSJSONSerialization.JSONObjectWithData(data!, options: []) as? NSDictionary {
+                    
+                    self._requestedWasOk = true
+                    
+                    self._exchangesDictionary = convertedJsonIntoDict["rates"] as? [String : Double]
+                    
+                    completion() //callback
+                    
+                }
+            } catch let error as NSError {
+                print(error.localizedDescription)
+                
+                self._requestedWasOk = false
+            }
+            
+        }
+        
+        task.resume()
+    }
+    
+    //displays equivalent values in different format currencies.
+    func updateValues() {
+     
+      dispatch_async(dispatch_get_main_queue(), {
+        
+            //Calculates and shows exchange value for UK Pounds
+            if var priceGBP = self._exchangesDictionary![CURRENCIES["UK"]!] {
+                priceGBP = priceGBP * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["UK"]!
+                self.lblGBP.text = self._currencyFormatter.stringFromNumber(priceGBP);
+            }
+        
+            //Calculates and shows exchange value for Euros
+            if var priceEUR = self._exchangesDictionary![CURRENCIES["EUR"]!] {
+                priceEUR = priceEUR * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["EUR"]!
+                self.lblEUR.text = self._currencyFormatter.stringFromNumber(priceEUR);
+                
+            }
+        
+            //Calculates and shows exchange value for Brasilian reals
+            if var priceBRL = self._exchangesDictionary![CURRENCIES["BRA"]!] {
+                priceBRL = priceBRL * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["BRA"]!
+                self.lblBRL.text = self._currencyFormatter.stringFromNumber(priceBRL);
+            }
+        
+            //Calculates and shows exchange value for Japan Yens
+            if var priceJPY = self._exchangesDictionary![CURRENCIES["JAP"]!] {
+                priceJPY = priceJPY * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["JAP"]!
+                self.lblJPY.text = self._currencyFormatter.stringFromNumber(priceJPY);
+            }
+      })
+    
+    }
+    
+    // detects when dollarNumberTextField changed
+    @IBAction func didDollarChange(sender: AnyObject) {
+        
```

Comment:
I'm confused by this method naming.   Are you asking a question?   If it's an action, then "dollarNumberTextFieldValueDidChange"